### PR TITLE
Use the correct span to rename after invoking extract-method manually.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://github.com/dotnet/roslyn/issues/58971

Test incoming.

### CODE REVIEW

File: src/EditorFeatures/Core/ExtractMethod/AbstractExtractMethodCommandHandler.cs
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/ExtractMethod/AbstractExtractMethodCommandHandler.cs:256]: the bug was here.  we were throwing away the second part of the result of GetFormattedDocumentASync.  That part of hte result tells us where the name token is to start inline-rename after the formatting occurs.

Comment (ryzngard) [reply src/EditorFeatures/Core/ExtractMethod/AbstractExtractMethodCommandHandler.cs:256]: Thanks for the explanation here :)

File: src/EditorFeatures/Core/ExtractMethod/AbstractExtractMethodCommandHandler.cs
Comment (ryzngard) [inline src/EditorFeatures/Core/ExtractMethod/AbstractExtractMethodCommandHandler.cs:154]: Do we ever expect this to be a default value? From code it is hard to tell

**Review by ryzngard — APPROVED** [review]
Approving with assumption of incoming test

File: src/EditorFeatures/Core/ExtractMethod/AbstractExtractMethodCommandHandler.cs
Comment (CyrusNajmabadi) [reply src/EditorFeatures/Core/ExtractMethod/AbstractExtractMethodCommandHandler.cs:154]: no.  it can't ever be a default value.  NOte: this is not assigned anything at the decl point.  It's purely a declaration of the variable.  there is one and only one value assignment to it from the method invoked.

Comment (ryzngard) [reply src/EditorFeatures/Core/ExtractMethod/AbstractExtractMethodCommandHandler.cs:154]: oh yea, I meant is it guaranteed to always be _something_ or do we have cases where the returned value is wrong? Not blocking on this PR, just curious

Comment (ryzngard) [reply src/EditorFeatures/Core/ExtractMethod/AbstractExtractMethodCommandHandler.cs:154]: We weren't using it before, so I could see that this might be an area for a bug we don't know about :) Don't spend time on it, just if you know off hand.

**Comment by Neme12** [comment]
I'm just surprised and curious, why was this added without a test? Adding it later is something that can be easily forgotten. Also, I don't think the issue should be marked as closed until there's a test that verifies the behavior.

**Comment by CyrusNajmabadi** [comment]
> I'm just surprised and curious, why was this added without a test? 

Because i'm working on tests as part of another PR in this area.

**Pull request merged**
